src/plugins/controls/public/control_types/options_list/options_list_editor.tsx

```diff
@@ -64,13 +66,54 @@ export const OptionsListEditor = ({
         dataView = await get(initialId);
       }
       if (!mounted) return;
-      setState((s) => ({ ...s, dataView, dataViewListItems }));
+      setState((s) => ({ ...s, dataView, dataViewListItems, fieldsMap: {} }));
     })();
     return () => {
       mounted = false;
     };
   });
 
+  useEffect(() => {
+    if (!state.dataView) return;
+
+    // double link the parent-child relationship so that we can filter in fields which are multi-typed to text / keyword
+    const doubleLinkedFields: OptionsListField[] = state.dataView?.fields.getAll();
+    for (const field of doubleLinkedFields) {
+      const parentFieldName = (field.subType as IFieldSubTypeMulti)?.multi?.parent;
+      if (parentFieldName) {
+        (field as OptionsListField).parentFieldName = parentFieldName;
+        const parentField = state.dataView?.getFieldByName(parentFieldName);
+        (parentField as OptionsListField).childFieldName = field.name;
+      }
+    }
+
+    const newFieldsMap: OptionsListEditorState['fieldsMap'] = {};
+    for (const field of doubleLinkedFields) {
+      if (field.type === 'boolean') {
+        newFieldsMap[field.name] = field;
+      }
+
+      // field type is keyword, check if this field is related to a text mapped field and include it.
+      if (field.aggregatable && field.type === 'string') {
```

Comment:
**Very** minor nit - it's technically a tiny bit more efficient to make this an `else if` since you don't have an early return after the `boolean` check above so, even if the field is a `boolean`, you still check if it's `aggregatable` for no reason. You could also add a `break` in that first `if`, but that's probably not great style.